crates/sui-indexer-alt-jsonrpc/src/api/transactions.rs

```diff
@@ -90,75 +89,11 @@ impl TransactionsApiServer for Transactions {
         digest: TransactionDigest,
         options: SuiTransactionBlockResponseOptions,
     ) -> RpcResult<SuiTransactionBlockResponse> {
-        use Error as E;
-
         let Self(ctx) = self;
 
-        let transaction = ctx.loader().load_one(TransactionKey(digest));
-        let balance_changes: OptionFuture<_> = options
-            .show_balance_changes
-            .then(|| ctx.loader().load_one(TxBalanceChangeKey(digest)))
-            .into();
-
-        let (transaction, balance_changes) = join!(transaction, balance_changes);
-
-        let transaction = transaction
-            .map_err(internal_error)?
-            .ok_or_else(|| invalid_params(E::NotFound(digest)))?;
-
-        // Balance changes might not be present because of pruning, in which case we return
-        // nothing, even if the changes were requested.
-        let balance_changes = match balance_changes.transpose().map_err(internal_error)? {
-            Some(None) => rpc_bail!(pruned("balance changes for transaction {digest}")),
-            Some(changes) => changes,
-            None => None,
-        };
-
-        let digest = TransactionDigest::try_from(transaction.tx_digest.clone())
-            .map_err(E::Conversion)
-            .map_err(internal_error)?;
-
-        let mut response = SuiTransactionBlockResponse::new(digest);
-
-        if options.show_input {
-            response.transaction = Some(
-                input_response(ctx, &transaction)
-                    .await
-                    .map_err(internal_error)?,
-            );
-        }
-
-        if options.show_raw_input {
-            response.raw_transaction = transaction.raw_transaction.clone();
-        }
-
-        if options.show_effects {
-            response.effects = Some(effects_response(&transaction).map_err(internal_error)?);
-        }
-
-        if options.show_raw_effects {
-            response.raw_effects = transaction.raw_effects.clone();
-        }
-
-        if options.show_events {
-            response.events = Some(
-                events_response(ctx, digest, &transaction)
-                    .await
-                    .map_err(internal_error)?,
-            );
-        }
-
-        if let Some(balance_changes) = balance_changes {
-            response.balance_changes =
-                Some(balance_changes_response(balance_changes).map_err(internal_error)?);
-        }
-
-        if options.show_object_changes {
-            response.object_changes =
-                Some(object_changes_response(ctx, digest, &transaction).await?);
-        }
-
-        Ok(response)
+        Ok(transaction_response(ctx, digest, options)
+            .await
+            .with_internal_context(|| format!("Failed to get transaction {digest}"))?)
```

Comment:
`with_internal_context` is used here instead of `internal_context` mainly because of its laziness, correct?